test/extended/networking/network_segmentation.go

```diff
@@ -682,6 +682,266 @@ var _ = Describe("[sig-network][OCPFeatureGate:NetworkSegmentation][Feature:User
 			expectedMessage := fmt.Sprintf("primary network already exist in namespace %q: %q", f.Namespace.Name, primaryNadName)
 			Expect(actualConditions[0].Message).To(Equal(expectedMessage))
 		})
+
+		Context("ClusterUserDefinedNetwork CRD Controller", func() {
+			const (
+				testClusterUdnName                = "test-cluster-net"
+				clusterUserDefinedNetworkResource = "clusteruserdefinednetwork"
+			)
+			var (
+				testTenantNamespaces []string
+			)
+			BeforeEach(func() {
+				testTenantNamespaces = []string{
+					f.Namespace.Name + "blue",
+					f.Namespace.Name + "red",
+				}
+
+				By("Creating test tenants namespaces")
+				for _, nsName := range testTenantNamespaces {
+					_, err := cs.CoreV1().Namespaces().Create(context.Background(), &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: nsName}}, metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					DeferCleanup(func() error {
+						err := cs.CoreV1().Namespaces().Delete(context.Background(), nsName, metav1.DeleteOptions{})
+						return err
+					})
+				}
+			})
+
+			BeforeEach(func() {
+				By("create test CR")
+				cleanup, err := createManifest("", newClusterUDNManifest(testClusterUdnName, testTenantNamespaces...))
+				DeferCleanup(func() error {
+					cleanup()
+					_, _ = e2ekubectl.RunKubectl("", "delete", clusterUserDefinedNetworkResource, testClusterUdnName)
```

Comment:
It may return not found error which I think should go silent. 
In follow-up commit there is additional logic to ensure the object is gone.
Make sense?

Reply:
Hmpf .... I don't like it. But... OK.

Let's at least be honest here and change the return type of the fn `DeferCleanup` invokes, ok ? I.e. no need to return an error since you don't do anything with it.

Reply:
Done

Reply:
Fixed in follow up commit